File: source/programing_exercises/nodejs/fizz_buzz.js
module.exports = class FizzBuzz {
    static generate(number) {
        let value = number;

        if (value % 3 === 0 && value % 5 === 0) {
            value = 'FizzBuzz'
        } else if (value % 3 === 0) {
            value = 'Fizz'
        } else if (value % 5 === 0) {
            value = 'Buzz'
        }

        return value;
    }

    static iterate(count) {
    const array = [];

    for (let i = 1; i <= count; i += 1) {
        array.push(FizzBuzz.generate(i));
    }

    return array;
    }
}

File: source/programing_exercises/python/20181018/fizz_buzz.py
class FizzBuzz():
    @staticmethod
    def generate(count):
        value = count

        if value % 3 == 0 and value % 5 == 0:
            value = 'FizzBuzz'
        elif value % 3 == 0:
            value = 'Fizz'
        elif value % 5 == 0:
            value = 'Buzz'

        return value        

    @staticmethod
    def iterate(count):
        array = []

        for n in range(count):
            array.append(FizzBuzz.generate(n + 1))

        return array
